lib/y2logsstep.pm

```diff
@@ -262,6 +273,27 @@ sub post_fail_hook {
     assert_script_run 'df > /tmp/df.txt';
     upload_logs '/tmp/df.txt';
 
+    $self->save_and_upload_log('pstree',  '/tmp/pstree');
+    $self->save_and_upload_log('ps auxf', '/tmp/ps_auxf');
+
+    # Collect yast2 installer trace
+    if (!script_run(qq{ps -eo pid,comm | grep -i [y]2start | cut -f 2 -d " " > /dev/$serialdev}, 0)) {
+        chomp(my $installer_pid = wait_serial(qr/^[\d{4}]/, 10));
+        my $trace_timeout = 120;
+        my $strace_ret = script_run("timeout $trace_timeout strace -f -o /installer_trace.log -tt -p $installer_pid", ($trace_timeout + 5));
+        # job time out -> EXIT_TIMEDOUT=124
+        ($strace_ret == 124) ? upload_logs '/installer_trace.log' : die 'strace did not timeout';
+
+        # collect installer proc fs files
+        foreach (@procfs_files) {
+            $self->save_and_upload_log("cat /proc/$installer_pid/$_", "/tmp/yast2-installer.$_");
+        }
+
+        assert_script_run 'extend gdb';
```

Comment:
Interesting, what is extend?

Reply:
Extension manager for Linuxrc.
https://github.com/yast/yast-installation/wiki/New-Installer:-Linuxrc-Extend

Reply:
wow, that's cool, something learned :)